client/src/components/player/PlayerControls.js

```diff
@@ -58,15 +60,104 @@ const PlayerPlaybackTime = styled.div`
 `
 
 class PlayerControls extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      progressBar: {
+        mouseDown: false
+      }
+    }
+
+    this.progressRef = React.createRef()
+  }
+
+  componentDidMount() {
+    this.getTotalTimeSong()
+    this.getCurrentTimeSong()
+    this.setMouseDownDefault()
+  }
+
+  setTimeSong = seconds => (this.props.audioRef.current.currentTime = seconds)
+
+  getTotalTimeSong = () => {
+    this.props.audioRef.current.addEventListener('canplay', () => {
+      const remaining = formatTime(this.props.audioRef.current.duration)
+
+      this.setState({
+        progressBar: {
+          ...this.state.progressBar,
+          remaining
+        }
+      })
+    })
+  }
+
+  getCurrentTimeSong = () => {
+    this.props.audioRef.current.addEventListener('timeupdate', () => {
+      if (this.props.audioRef.current.duration) {
+        this.setState({
+          progressBar: {
+            ...this.state.progressBar,
+            current: formatTime(this.props.audioRef.current.currentTime),
+            percentage:
+              (this.props.audioRef.current.currentTime /
+                this.props.audioRef.current.duration) *
+              100
+          }
+        })
+      }
+    })
+  }
+
+  onClickChangeProgressBar = e => {
+    const percentage =
+      (e.nativeEvent.offsetX / this.progressRef.current.offsetWidth) * 100
+    const seconds = this.props.audioRef.current.duration * (percentage / 100)
+
+    this.setTimeSong(seconds)
+  }
+
+  onMouseMoveChangeProgressBar = e => {
+    if (this.state.progressBar.mouseDown) {
+      const percentage =
+        (e.nativeEvent.offsetX / this.progressRef.current.offsetWidth) * 100
+      const seconds = this.props.audioRef.current.duration * (percentage / 100)
+
+      this.setTimeSong(seconds)
+    }
+  }
+
+  onMouseDownChangeProgressBar = () => this.toggleMouseDownProgressBar()
```

Comment:
Im not sure why these two are needed, you could pass `this. toggleMouseDownProgressBar()` as a callback, no need to wrap it